src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -954,7 +950,9 @@ class Namer { typer: Typer =>
       //
       // The scheme critically relies on an implementation detail of isRef, which
       // inspects a TypeRef's info, instead of simply dealiasing alias types.
-    val rhsType = abstracted(typedAheadType(tdef.rhs).tpe)
+
+    val isDerived = tdef.rhs.isInstanceOf[untpd.DerivedTypeTree]
+    val rhsType = abstracted(typedAheadType(tdef.rhs).tpe, canAbstract = !isDerived)
```

Comment:
Having a boolean parameter that changes the behavior of a method to become the identity seems weird to me, I would just write:
```scala
val rhsType0 = typedAheadType(tdef.rhs).tpe
val rhsType = if (!isDerived) abstracted(rhsType0) else rhsType0
```